### PR TITLE
quincy: test/bufferlist: ensure rebuild_aligned_size_and_memory() always rebuilds.

### PR DESCRIPTION
**Pull request by cbodley**
backport tracker: https://tracker.ceph.com/issues/53974

clean cherry-pick required additional commit from https://github.com/ceph/ceph/pull/44675

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by cbodley** [comment]
@neha-ojha @rzarzynski would like to see this merge asap, as the bufferlist failure is blocking merge of other quincy PRs

**Review by rzarzynski — APPROVED** [review]

**Comment by cbodley** [comment]
p.s. thanks for the fix Radek <3

**Comment by rzarzynski** [comment]
Thanks for the backport, Casey! If there are not objections, I will merge it today, just after the big green button becomes available. It's a unittest fix, so I doubt we need a teuthology run.

**Comment by neha-ojha** [comment]
> Thanks for the backport, Casey! If there are not objections, I will merge it today, just after the big green button becomes available. It's a unittest fix, so I doubt we need a teuthology run.

go for it

**Review by idryomov — APPROVED** [review]

**Pull request merged**
